Sources/Ledger/PartialTxOut.swift

```diff
@@ -22,23 +23,29 @@ extension PartialTxOut {
             encryptedMemo: txOut.encryptedMemo,
             commitment: txOut.commitment,
             maskedValue: txOut.maskedValue,
+            maskedTokenId: txOut.maskedTokenId,
             targetKey: txOut.targetKey,
             publicKey: txOut.publicKey)
     }
 }
 
 extension PartialTxOut {
     init?(_ txOut: External_TxOut) {
-        guard let commitment = Data32(txOut.amount.commitment.data),
+        guard let commitment = Data32(txOut.maskedAmount.commitment.data),
               let targetKey = RistrettoPublic(txOut.targetKey.data),
               let publicKey = RistrettoPublic(txOut.publicKey.data)
         else {
             return nil
         }
+        guard [0,4,8].contains(txOut.maskedAmount.maskedTokenID.count) else {
```

Comment:
why not just add this to the long list of prior guard statements?

Reply:
ya not sure, ill move it up